Source/Details/_ASCollectionReusableView.h

```diff
@@ -18,15 +18,41 @@
 #import <UIKit/UIKit.h>
 #import <AsyncDisplayKit/ASBaseDefines.h>
 
-@class ASCellNode, ASCollectionElement;
+@class ASCollectionElement;
+
+/**
+ * Attempts to cast x to _ASCollectionReusableView, or nil if not possible.
+ *
+ * Since _ASCollectionReusableView is not available for subclassing (see below),
+ * comparing x's and _ASCollectionReusableView's classes is faster than calling -isKindOfClass: on x.
+ */
+#define ASCollectionReusableViewCast(x) ({ \
+  id __var = x; \
+  ((_ASCollectionReusableView *) (x.class == [_ASCollectionReusableView class] ? __var : nil)); \
+})
+
+/**
+ * Attempts to cast x to _ASCollectionReusableView and assigns to __var. If not possible, returns the given __val.
+ *
+ * Since _ASCollectionReusableView is not available for subclassing (see below),
+ * comparing x's and _ASCollectionReusableView's classes is faster than calling -isKindOfClass: on x.
+ */
+#define ASCollectionReusableViewCastOrReturn(x, __var, __val) \
+  _ASCollectionReusableView *__var = ASCollectionReusableViewCast(x); \
+  if (__var == nil) { \
+    return __val; \
+  }
 
 NS_ASSUME_NONNULL_BEGIN
 
 AS_SUBCLASSING_RESTRICTED
+
 @interface _ASCollectionReusableView : UICollectionReusableView
-@property (nonatomic, strong, readonly, nullable) ASCellNode *node;
-@property (nonatomic, strong, nullable) ASCollectionElement *element;
+
 @property (nonatomic, strong, nullable) UICollectionViewLayoutAttributes *layoutAttributes;
+
+- (void)setElement:(ASCollectionElement *)element;
```

Comment:
I don't totally agree with removing the readwrite property. It's true that the visibleMap is the source of truth about what ought to be displayed, but it is useful to be able to read, from a view, what element it is currently displaying in reality. For example, to be able to tell in `collectionView:cellForItemAtIndexPath:` what element the cell is transitioning from, or to read for debugging purposes. And since these classes are underbarred, I think the risk of external clients leaning on the property is really minor.